src/compiler/utilities.ts

```diff
@@ -2566,6 +2566,10 @@ namespace ts {
             || kind === SyntaxKind.NumericLiteral;

```

Comment:
what happens if you add `|| isNoSubstitutionTemplateLiteral(node)` directly to `isStringOrNumericLiteral` ? Maybe you can just do that plus maybe change the name to be more accurate.

Reply:
The only remaining uses are in `getPropertyNameOfBindingOrAssignmentElement`. That function has no return type, but  changing it and recompiling shows a number of errors. Looks like the return type does need to be `PropertyName | undefined` which excludes `NoSubstitutionTemplateLiteral`.